test/externalTests/plugins/testCommon.js

```diff
@@ -77,136 +80,116 @@ function inject (bot) {
-      setInventorySlot(36, new Item(item.type, 1, 0), () => {
-        placeBlock(36, position, resume)
-      })
-    }
-
-    function resume () {
-      resetBlocksToSuperflat(cb)
+      await setInventorySlot(36, new Item(item.type, 1, 0))
+      await placeBlock(36, position)
     }
   }
 
-  function placeBlock (slot, position, cb) {
+  async function placeBlock (slot, position) {
     bot.setQuickBarSlot(slot - 36)
     // always place the block on the top of the block below it, i guess.
     const referenceBlock = bot.blockAt(position.plus(new Vec3(0, -1, 0)))
-    bot.placeBlock(referenceBlock, new Vec3(0, 1, 0), cb)
+    return bot.placeBlock(referenceBlock, new Vec3(0, 1, 0))
   }
 
   // always leaves you in creative mode
-  function resetState (cb) {
-    callbackChain([
-      becomeCreative,
-      clearInventory,
-      (cb) => {
-        // console.log('start flying')
-        bot.creative.startFlying()
-        teleport(new Vec3(0, 4, 0), cb)
-      },
-      cb => bot.waitForChunksToLoad(cb),
-      resetBlocksToSuperflat,
-      (cb) => { setTimeout(cb, 1000) },
-      clearInventory
-    ], cb)
+  async function resetState () {
+    await becomeCreative()
+    await clearInventory()
+    bot.creative.startFlying()
+    await teleport(new Vec3(0, 4, 0))
+    await bot.waitForChunksToLoad()
+    await resetBlocksToSuperflat()
+    await sleep(1000)
+    await clearInventory()
   }
 
-  function becomeCreative (cb) {
+  async function becomeCreative () {
     // console.log('become creative')
-    setCreativeMode(true, cb)
+    return setCreativeMode(true)
   }
 
-  function becomeSurvival (cb) {
-    setCreativeMode(false, cb)
+  async function becomeSurvival () {
+    return setCreativeMode(false)
   }
 
-  function setCreativeMode (value, cb) {
+  async function setCreativeMode (value) {
     // this function behaves the same whether we start in creative mode or not.
     // also, creative mode is always allowed for ops, even if server.properties says force-gamemode=true in survival mode.
 
-    function onMessage (jsonMsg) {
+    const onMessage = (jsonMsg) => {
       // console.log(jsonMsg)
       switch (jsonMsg.translate) {
         case 'commands.gamemode.success.self':
         case 'gameMode.changed':
-          // good.
-          bot.removeListener('message', onMessage)
-          clearTimeout(timeOut)
-          return cb()
+          return true
         case 'commands.generic.permission':
           sayEverywhere('ERROR: I need to be an op (allow cheats).')
-          bot.removeListener('message', onMessage)
         // at this point we just wait forever.
         // the intention is that someone ops us while we're sitting here, then you kill and restart the test.
       }
       // console.log("I didn't expect this message:", jsonMsg);
+      return false
     }
-    bot.on('message', onMessage)
+
+    // For whatever reason, timeout = success, and there is no other failure condition in this promise
```

Comment:
I guess, that's not intended. Should probably throw an error..

Reply:
If I throw an error, tests start failing. I would agree this should throw an error, but IMO it is out of scope of this PR as this is the same behavior that has always been here.